### PR TITLE
feat: Archive and Trash Notes Functionality Provided

### PR DESCRIPTION
**Pull request by dev-enforced**
In this Pull Request and the Live Preview which the reviewer will see the reviewer will be able to generate notes and be able to shift them to archives and trash sections respectively. And from the archive and trash sections one can restore the notes back to their original state.

List of Files to go through:
1) src/context/TrashContext.jsx
2) src/context/ArchivesContext.jsx

API calls made in services folder having the subfolders related to :
1) Archives
2) Notes
3) Trash

Components Built up for this purpose include:
1) ArchivedNoteItem folder
2) TrashNoteItem folder 
3) NoteItem folder 

Pages built up for this purpose include:
 1) NotesArchives folder having empty and filled archive sections depending on the state of archives 
2) NotesTrash folder having empty and filled archive sections depending on the state of trash

**Backend Folder does not need to be reviewed for this PR**

### CODE REVIEW

File: src/constants/icons.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { MdDarkMode as Dark } from "react-icons/md";
+import { MdDarkMode as Dark, MdUnarchive as Unarchive } from "react-icons/md";
 import { BiPin as Unpinned, BiPen as Edit, BiArchiveIn as ArchiveMoved, BiTrashAlt as Trash } from "react-icons/bi";
 import { BsPinFill as Pinned } from "react-icons/bs"
-export { Dark, Unpinned, Pinned, ArchiveMoved, Trash, Edit };
\ No newline at end of file
+export { Dark, Unpinned, Pinned, ArchiveMoved, Trash, Edit, Unarchive };
```
Comment (Namrata-J) [inline src/constants/icons.js]: liked the way you are extracting icons in a different file!!

File: src/pages/NotesArchives/FilledNotesArchives.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import "./NotesArchives.css";
+
+const FilledNotesArchives = ({displayList}) => {
+    return (
+        <>
+            <h4 className='text-center fw-700'>ARCHIVED NOTES</h4>
+            <div className="gentle-grid-responsive py-8 m-4">
+                {/* {
```
Comment (Namrata-J) [inline src/pages/NotesArchives/FilledNotesArchives.jsx]: remove this commented code if not needed!!

**Review by Namrata-J — APPROVED** [review]

File: src/pages/NotesArchives/FilledNotesArchives.jsx (hunk 2 above)
Comment (dev-enforced) [reply src/pages/NotesArchives/FilledNotesArchives.jsx]: I don't know why this snippet of code appears to be commented when I have uncommented it in my local 
https://github.com/dev-enforced/Notes-App-Recreated/blob/archive-and-trash-operations/src/pages/NotesArchives/FilledNotesArchives.jsx
On Github also it shows to be uncommented:

**Review by sakshi006 — APPROVED** [review]
1. You can remove archiveNoteitem.css and trashNoteItem,css files as they are empty
2. The code is clean and well structured.
3.  I liked how for every service/function, you have created a different file, makes the code look cleaner.

**Review by ankit9015 — APPROVED** [review]
PR is good to merge. 
1. I like how you are importing icons from a single file 
2. The folders are nicely structured
3. I like how you are using index.js in different folders to export relevant code

**Pull request merged**